### PR TITLE
DAOS-2746 test: Add junit-capable go test runner

### PR DESCRIPTION
**Pull request by mjmac**
The gotestsum runner has a --junitfile option that will
convert the native runner's JSON output into JUnit for
consumption by Jenkins.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -887,11 +888,15 @@ pipeline {
                                                    mv $DAOS_BASE/install/bin/daos_admin \
                                                       $DAOS_BASE/install/bin/orig_daos_admin
 
+                                               # install junit-capable go test runner
+                                               curl -sSL "https://github.com/gotestyourself/gotestsum/releases/download/v${GTS_RELEASE}/gotestsum_${GTS_RELEASE}_linux_amd64.tar.gz" | \
```
Comment (brianjmurrell) [inline Jenkinsfile]: Should add `-f` to your `curl` command options.

Comment (mjmac) [reply Jenkinsfile]: Well, if you really want to build from scratch, the project does publish source tarballs, e.g. [v0.4.0](https://github.com/gotestyourself/gotestsum/archive/v0.4.0.tar.gz).

I would imagine that there are rpm macros for go builds now, so it shouldn't be too hard to make an RPM out of this.

**Comment by mjmac** [comment]
Going to hold off on merging this until https://github.com/daos-stack/gotestsum/pull/2 lands, as it doesn't make sense to introduce a new source of potential test failures with the curl download.

**Comment by mjmac** [comment]
OK, this is finally ready for re-review and hopefully landing. I added the pragmas to skip functional/hardware tests as this code doesn't touch any of that stuff.

We can see that the Go tests results are correctly reported in JUnit format, e.g. https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-1818/17/testReport/(root)/server_ioserver/

**Comment by mjmac** [comment]
@brianjmurrell, @jolivier23: You guys good with this PR now? Moving gotestsum into an RPM makes this even simpler. The script will prefer to use gotestsum but will fall back on the old test runner if it has to so that it still works for developers who haven't installed the new runner.

**Pull request merged**